src/cql/r4/MMECalculator.cql

```diff
@@ -66,7 +66,14 @@ define function GetMedicationQuantity(medication MedicationRequest):
       ingredient: OMTKLogic.GetIngredients(rxNormCode),
       prescription: if ingredient is not null then First(ingredient) else null,
       dosageInstruction: O.dosageInstruction[0],
-      dispenseRequestQuantity: O.dispenseRequest.quantity.value.value,
+      dispenseRequest: O.dispenseRequest,
+      supplyDuration: if dispenseRequest.expectedSupplyDuration is not null then System.Quantity {
+        value: dispenseRequest.expectedSupplyDuration.value,
+        unit: dispenseRequest.expectedSupplyDuration.unit
+      } else null as System.Quantity,
+      dispenseRequestQuantity: dispenseRequest.quantity.value.value,
+      //get dose quantity daily
+      dispenseRequestDoseQuantity: if supplyDuration is not null then dispenseRequestQuantity / supplyDuration.value else null as System.Quantity,
       doseAndRate:  if dosageInstruction is not null then dosageInstruction.doseAndRate[0] else null, //either a single doseRange or doseQuantity see specs here:  https://www.hl7.org/fhir/dosage-definitions.html#Dosage.doseAndRate.dose_x_
```

Comment:
I tried changing `dosageInstructions.doseAndRate.doseQuantity.value` from `1` to different values and I never saw the daily MME change. 

Is `dosageInstructions.doseAndRate.doseQuantity.value` being checked before falling back to `dispenseRequestQuantity / supplyDuration.value`?

This is not necessarily a problem- I would rather have more of the MME calculation in the MME calculator, than relying on the backend to add extra data

Reply:
@ivan-c It looks for `doseInstruction.doseAndRate.doseQuantity.value first` **first** before falling back.  As I have commented, I think the issue is that the MedicationRequest Json returned from our confidential backend is putting `dosageInstruction` under `dispenseRequest` when it shouldn't be.  I think dosageInstruction is a **sibling** of dispenseRequest not a **child** of dispenseRequest in the JSON structure. See here:  https://www.hl7.org/fhir/medicationrequest.htmlhttps://www.hl7.org/fhir/medicationrequest.html
This is the snippet of JSON that works:

```{
      "resource": {
        "authoredOn": "2020-10-25",
        "id": "3e14d51b-58d8-4e7a-b9e3-26893e5604d2",
        "subject": {
          "reference": "Patient/5c41cecf-cf81-434f-9da7-e24e5a99dbc2"
        },
        "dosageInstruction": [
          {
            "doseAndRate": [
              {
                "doseQuantity": {
                  "value": 2.0
                }
              }
            ],
            "timing": {
              "repeat": {
                "frequency": 15.0,
                "period": 15.0,
                "periodUnit": "d"
              }
            }
          }
        ],
        "dispenseRequest": {
          "expectedSupplyDuration": {
            "code": "d",
            "system": "http://unitsofmeasure.org",
            "unit": "days",
            "value": 15.0
          },
          "extension": [
            {
              "url": "http://cosri.org/fhir/pharmacy_name",
              "valueString": "TEST, DOCTOR"
            },
            {
              "url": "http://cosri.org/fhir/last_fill",
              "valueDate": "2020-10-25"
            }
          ],
          "quantity": {
            "value": 30.0
          }
        },
        "identifier": [
          {
            "system": "https://github.com/uwcirg/script-fhir-facade/"
          }
        ],
        "medicationCodeableConcept": {
          "coding": [
            {
              "code": "49999084830",
              "display": "morphine sulfate 15 MG Oral Tablet",
              "system": "http://hl7.org/fhir/sid/ndc"
            },
            {
              "code": "892582",
              "display": "morphine sulfate 15 MG Oral Tablet",
              "system": "http://www.nlm.nih.gov/research/umls/rxnorm"
            }
          ],
          "text": "morphine sulfate 15 MG Oral Tablet"
        },
        "requester": {
          "display": "TEST PRESCRIBER"
        },
        "resourceType": "MedicationRequest"
      }
    }```

Reply:
@ivan-c just for context, the CQL compilation resulted in error if I were to look for dosageInstruction under dispenseRequest, i.e. `dispenseRequest.doseInstruction`:
![Screen Shot 2021-06-07 at 9 23 46 AM](https://user-images.githubusercontent.com/12942714/121055550-35569d00-c772-11eb-96b9-33723d3725fc.png)

Reply:
Thanks @achen2401 I had missed that I needed to fix the error in the nesting of `dosageInstruction`; I've got that fixed locally, should have a PR shortly